src/Spreadsheet.tsx

```diff
@@ -30,89 +25,99 @@ import * as Actions from "./actions";
 import "./Spreadsheet.css";
 
 type DefaultCellType = {
-  value: string | number | boolean | null
+  value: string | number | boolean | null;
 };
 
-const getValue = ({ data }: { data: ?DefaultCellType }) =>
+const getValue = ({ data }: { data?: DefaultCellType }) =>
   data ? data.value : null;
 
-export type Props<CellType: Types.CellBase, Value> = {|
-  data: Matrix.Matrix<CellType>,
-  columnLabels?: string[],
-  ColumnIndicator?: ComponentType<ColumnIndicatorProps>,
-  rowLabels?: string[],
-  RowIndicator?: ComponentType<RowIndicatorProps>,
-  hideRowIndicators?: boolean,
-  hideColumnIndicators?: boolean,
-  Table: ComponentType<TableProps>,
-  Row: ComponentType<RowProps>,
-  Cell: ComponentType<CellProps<CellType, Value>>,
-  DataViewer: Types.DataViewer<CellType, Value>,
-  DataEditor: Types.DataEditor<CellType, Value>,
-  onKeyDown?: (event: SyntheticKeyboardEvent<HTMLElement>) => void,
-  getValue: Types.getValue<CellType, Value>,
-  getBindingsForCell: Types.getBindingsForCell<CellType>,
-  store: Store
-|};
-
-type Handlers = {|
-  cut: () => void,
-  copy: () => void,
-  paste: () => void,
-  setDragging: boolean => void,
-  onKeyDownAction: (SyntheticKeyboardEvent<HTMLElement>) => void,
-  onKeyPress: (SyntheticKeyboardEvent<HTMLElement>) => void,
-  onDragStart: () => void,
-  onDragEnd: () => void
-|};
-
-type State = {|
-  rows: number,
-  columns: number,
-  mode: Types.Mode
-|};
+export type Props<CellType, Value> = {
+  data: Matrix.Matrix<CellType>;
+  columnLabels?: string[];
+  ColumnIndicator?: React.FC<ColumnIndicatorProps>;
+  rowLabels?: string[];
+  RowIndicator?: React.FC<RowIndicatorProps>;
+  hideRowIndicators?: boolean;
+  hideColumnIndicators?: boolean;
+  Table: React.FC<TableProps>;
+  Row: React.FC<RowProps>;
+  Cell: React.FC<CellProps<CellType, Value>>;
+  DataViewer: Types.DataViewer<CellType, Value>;
+  DataEditor: Types.DataEditor<CellType, Value>;
+  onKeyDown?: (event: React.KeyboardEvent<HTMLInputElement>) => void;
+  getValue: Types.getValue<CellType, Value>;
+  getBindingsForCell: Types.getBindingsForCell<CellType>;
+  store: Store<any>;
+};
+
+type Handlers = {
+  cut?: () => void;
+  copy?: () => void;
+  paste?: () => void;
+  setDragging?: (_: boolean) => void;
+  onKeyDownAction?: (_: React.SyntheticEvent<HTMLElement>) => void;
+  onKeyPress?: (_: React.SyntheticEvent<HTMLElement>) => void;
+  onDragStart?: () => void;
+  onDragEnd?: () => void;
+};
+
+type State = {
+  rows: number;
+  columns: number;
+  mode: Types.Mode;
+};
 
 type ColumnIndicatorProps = {
-  column: number,
-  label?: Node | null
+  column: number;
+  label?: string | null;
```

Comment:
Unclear why this was narrowed to `string` - why can't I pass a JSX element here?
```suggestion
  label?: ReactNode | null;
```